### PR TITLE
[tempest]Enable live-migration tests

### PR DESCRIPTION
**Pull request by gibizer**
Depends-On: https://github.com/openstack-k8s-operators/ci-framework/pull/941

### CODE REVIEW

**Review by SeanMooney — APPROVED** [review]
this looks correct and all the deps should be met already

so approving. if this does not work for some reason it will be blocked by the zuul job that is currently running

**Comment by gibizer** [comment]
Sad face. It seems we have some DNS issues on the compute

```
{'id': 1, 'source_compute': 'compute-2.ci-rdo.local', 'dest_compute': 'compute-1.ci-rdo.local', 'source_node': 'compute-2.ci-rdo.local', 'dest_node': 'compute-1.ci-rdo.local', 'dest_host': None, 'old_instance_type_id': 1, 'new_instance_type_id': 1, 'instance_uuid': 'f9989378-588b-44ce-a28f-450dd7eea46a', 'status': 'failed', 'migration_type': 'live-migration', 'created_at': '2023-12-07T14:31:47.000000', 'updated_at': '2023-12-07T14:31:53.000000'}]
```

```
2023-12-07 14:31:52.903 2 ERROR oslo_messaging.rpc.server Command: scp -C -r compute-2.ci-rdo.local:/var/lib/nova/instances/f9989378-588b-44ce-a28f-450dd7eea46a/disk.config /var/lib/nova/instances/f9989378-588b-44ce-a28f-450dd7eea46a
2023-12-07 14:31:52.903 2 ERROR oslo_messaging.rpc.server Exit code: 255
2023-12-07 14:31:52.903 2 ERROR oslo_messaging.rpc.server Stdout: ''
2023-12-07 14:31:52.903 2 ERROR oslo_messaging.rpc.server Stderr: 'ssh: Could not resolve hostname compute-2.ci-rdo.local: Name or service not known\r\nConnection closed\r\n'
2023-12-07 14:31:52.903 2 ERROR oslo_messaging.rpc.server 
```

**Comment by gibizer** [comment]
recheck 
lets see if https://github.com/openstack-k8s-operators/ci-framework/pull/917 helps

**Comment by gibizer** [comment]
https://github.com/openstack-k8s-operators/ci-framework/pull/917 helps a bit but it is not the full solution. Now the DNS issue is resolved, but we fail at host key verification:
```
2023-12-08 13:02:13.630 2 ERROR oslo_messaging.rpc.server oslo_concurrency.processutils.ProcessExecutionError: Unexpected error while running command.
2023-12-08 13:02:13.630 2 ERROR oslo_messaging.rpc.server Command: scp -C -r compute-1.ci-rdo.local:/var/lib/nova/instances/84f94f22-534f-400b-a723-193d3570dd8f/disk.config /var/lib/nova/instances/84f94f22-534f-400b-a723-193d3570dd8f
2023-12-08 13:02:13.630 2 ERROR oslo_messaging.rpc.server Exit code: 255
2023-12-08 13:02:13.630 2 ERROR oslo_messaging.rpc.server Stdout: ''
2023-12-08 13:02:13.630 2 ERROR oslo_messaging.rpc.server Stderr: 'Host key verification failed.\r\nConnection closed\r\n'
2023-12-08 13:02:13.630 2 ERROR oslo_messaging.rpc.server 
```

The known_hosts file still does not talk about the proper hostname
https://review.rdoproject.org/zuul/build/e7f1515e56894b7e9bab34cafb219975/log/controller/ci-framework-data/logs/192.168.122.100/etc/ssh/ssh_known_hosts
```
edpm-compute-1.ctlplane.example.com,192.168.122.101,edpm-compute-1* ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQCYRIQqy7Qnw/dZS2q7lAJp/JXzAFnLHpGQYKblrWuisLuT+MJ8olfMZbBUTSLuO66T0d2NKbdke1x1u8OiJeCOBnrDd4ZpR7pnEsVYN/3XrLJfAN/edD+ttpGPi8yGwQADujYzdh3VJRKhNvWxfYFYNPLwGYLDoFJuAedwcWD80Tqv4uOI7WPU9w8kuw/YuLhxpUDAiirnt3Ratr0wfrAJFoFeUJzRJXABmm/p4p20BH3WraCu1huyxU29T7ZRXxomMM684sM2uunvjj+dQM6Z3R6i4pse7WVTxwzS9h2L2K702zKbR1txBwgmUVE4W0zEJKC/mHnoKjZjjPhsKsfUbYF7T2FZ1tNUvoX4td42Ld9Oit+MoBr9aiCucjUNhI52OVFgen1iEMMOA6Ma+jhjdJUPa2gH7g2eBHvntjAVVTzIKmZzGMuWn8DW3oSWmfdUB1GHViXMiqd3n76503LV7onihoiE/dRX0td0dBLbJT3SmX1FbJIdBTiSkzYl8Hc=
```

**Comment by rabi** [comment]
recheck

**Comment by gibizer** [comment]
It seems https://github.com/openstack-k8s-operators/dataplane-operator/pull/567 helps with the hostname issues in CI. Thanks @rabi 

The current error is
```
2023-12-09 05:04:26.882 2 ERROR nova.virt.libvirt.driver [None req-b6905121-f659-4817-880d-245657552fda 1dd6c5b7a68e49acafdca4abce71b91f 7391759ef72841fc9ecf4dac68f0ed29 - - default default] [instance: 06450674-7fa4-4e27-af80-e28def7a9ac5] Live Migration failure: operation failed: guest CPU doesn't match specification: missing features: xsaves: libvirt.libvirtError: operation failed: guest CPU doesn't match specification: missing features: xsaves
```
https://review.rdoproject.org/zuul/build/53352d02b1bb4b5aadeb783c088567ce/log/controller/ci-framework-data/logs/192.168.122.102/log/containers/nova/nova-compute.log#5959

**Comment by gibizer** [comment]
/hold
We need a CI specific nova config to allow live migration between computes with different CPU flags.

**Comment by gibizer** [comment]
recheck CI framework dependency is updated

**Comment by gibizer** [comment]
recheck CI framework dependency is updated

**Comment by gibizer** [comment]
recheck CI framework dependency is updated

**Comment by gibizer** [comment]
We have one successful live migration \o/
```
{1} tempest.api.compute.admin.test_live_migration.LiveAutoBlockMigrationV225Test.test_live_migration_with_trunk [38.007063s] ... ok
```
* I need to filter the live migration tests that require shared storage / volume booted instance. 
* I need to check that why the rest of the LiveAutoBlockMigrationV225Test skipped.

File: .zuul.yaml [hunk 1]
```diff
@@ -167,11 +167,15 @@
...
-        - tempest.api.compute.admin.test_live_migration
-        - test_server_connectivity_live_migration
       # Swift test failing with unauthorized errors
         - tempest.scenario.test_object_storage
+      # We need to use a custom cpu model to allow live migrating between
+      # slightly different computes coming from the node pool
+      cifmw_edpm_deploy_nova_compute_extra_config: |
+        [libvirt]
+        cpu_mode = custom
+        cpu_models = Nehalem
```
Comment (gibizer) [inline .zuul.yaml:177]: Passed all the way down to the compute service https://review.rdoproject.org/zuul/build/15d16f851b4c4b7492b130a68048cba5/log/controller/ci-framework-data/logs/192.168.122.100/log/containers/nova/nova-compute.log#466

**Comment by gibizer** [comment]
> We have one successful live migration \o/
> 
> ```
> {1} tempest.api.compute.admin.test_live_migration.LiveAutoBlockMigrationV225Test.test_live_migration_with_trunk [38.007063s] ... ok
> ```
> 
> * I need to filter the live migration tests that require shared storage / volume booted instance.
Done
> * I need to check that why the rest of the LiveAutoBlockMigrationV225Test skipped.
We needed yet another feature flag enable for them. Done.

**Comment by gibizer** [comment]
I need one more flag and one more filter :)

**Comment by gibizer** [comment]
Live migration from paused state failed:
```
testtools.matchers._impl.MismatchError: 'compute-1.ci-rdo.local' != 'compute-0.ci-rdo.local': Live Migration failed. Migrations list for Instance 88084fdd-51dc-4967-8493-c1169e4533ec: [
{'id': 4, 'source_compute': 'compute-0.ci-rdo.local', 'dest_compute': 'compute-1.ci-rdo.local', 'source_node': 'compute-0.ci-rdo.local', 'dest_node': 'compute-1.ci-rdo.local', 'dest_host': None, 'old_instance_type_id': 1, 'new_instance_type_id': 1, 'instance_uuid': '88084fdd-51dc-4967-8493-c1169e4533ec', 'status': 'failed', 'migration_type': 'live-migration', 'created_at': '2023-12-16T11:07:37.000000', 'updated_at': '2023-12-16T11:07:45.000000'}]
```

source compute:
```
2023-12-16 11:07:45.380 2 ERROR nova.virt.libvirt.driver [None req-4e14c92b-915c-4df2-ac6f-467686624677 c975637b06c046b3b17ec0b2db35f3e8 2faa4b99766143de9c8527f163dc0a72 - - default default] [instance: 88084fdd-51dc-4967-8493-c1169e4533ec] Live Migration failure: End of file while reading data: : Input/output error: libvirt.libvirtError: End of file while reading data: : Input/output error
```
https://review.rdoproject.org/zuul/build/3967185da54e4a96904ab3c7678548a3/log/controller/ci-framework-data/logs/192.168.122.100/log/containers/nova/nova-compute.log#6775

Hmm it sounds like a nova bug to try to live migrate a pause VM with post-copy.
```
2023-12-16 11:07:45.356+0000: 102512: debug : qemuDomainAssignUSBAddresses:3254 : Finished assigning USB ports
2023-12-16 11:07:45.356+0000: 102512: debug : qemuMigrationSrcPerformPeer2Peer3:5585 : Prepare3 0x7f4ae0013d90
2023-12-16 11:07:45.356+0000: 102512: debug : qemuDomainObjEnterRemote:6421 : Entering remote (vm=0x7f4b0800e850 name=instance-0000000a)
2023-12-16 11:07:45.370+0000: 102512: error : virNetClientProgramDispatchError:172 : argument unsupported: post-copy migration is not supported with non-live or paused migration
2023-12-16 11:07:45.370+0000: 102512: debug : qemuDomainObjExitRemote:6432 : Exited remote (vm=0x7f4b0800e850 name=instance-0000000a)
2023-12-16 11:07:45.370+0000: 102512: debug : virConnectIsAlive:1450 : conn=0x7f4ae0013d90
2023-12-16 11:07:45.370+0000: 102512: debug : qemuDomainObjEnterRemote:6421 : Entering remote (vm=0x7f4b0800e850 name=instance-0000000a)
2023-12-16 11:07:45.370+0000: 102512: debug : virConnectUnregisterCloseCallback:1538 : conn=0x7f4ae0013d90
2023-12-16 11:07:45.371+0000: 102512: debug : qemuDomainObjExitRemote:6432 : Exited remote (vm=0x7f4b0800e850 name=instance-0000000a)
2023-12-16 11:07:45.378+0000: 102512: error : virNetSocketReadWire:1780 : End of file while reading data: : Input/output error
```
https://logserver.rdoproject.org/21/621/fed3ba3a759026a56618d80c87873492405f33e4/github-check/nova-operator-tempest-multinode/3967185/controller/ci-framework-data/logs/192.168.122.100/log/libvirt/virtqemud.log

**Comment by gibizer** [comment]
> Live migration from paused state failed:
> ...

Temporary disabled test_live_block_migration_paused as there is probably an upstream bug with post-copy live migration of a paused instance.

**Comment by gibizer** [comment]
\o/ we have a full green CI with passing live migration tests cases:
* test_live_block_migration[id-1dce86b8-eb04-4c03-a9d8-9c1dc3ee0c7b,multinode]
* test_live_migration_with_trunk[id-0022c12e-a482-42b0-be2d-396b5f0cffe3,multinode,network]

**Comment by gibizer** [comment]
> > Live migration from paused state failed:
> > ...
> 
> Temporary disabled test_live_block_migration_paused as there is probably an upstream bug with post-copy live migration of a paused instance.

Found an upstream bug https://bugs.launchpad.net/nova/+bug/1946752

**Comment by gibizer** [comment]
/unhold 
dependency landed

**Review by SeanMooney — APPROVED** [review]

**Pull request merged**